switch/physfs/physfs.patch

```diff
@@ -0,0 +1,191 @@
++#else
+     st->readonly = (access(fname, W_OK) == -1);
++#endif
+     return 1;
+ } /* __PHYSFS_platformStat */
+ 
+diff -Naur physfs-3.0.2/src/physfs_platforms.h physfs-3.0.2-patched/src/physfs_platforms.h
+--- physfs-3.0.2/src/physfs_platforms.h	2020-05-21 14:48:36.117670131 +0100
++++ physfs-3.0.2-patched/src/physfs_platforms.h	2020-05-21 14:56:30.372458194 +0100
+@@ -72,6 +72,10 @@
+ #elif defined(unix) || defined(__unix__)
+ #  define PHYSFS_PLATFORM_UNIX 1
+ #  define PHYSFS_PLATFORM_POSIX 1
++#elif defined(__SWITCH__)
++#  define PHYSFS_PLATFORM_POSIX 1
++#  define PHYSFS_PLATFORM_SWITCH 1
++#  define PHYSFS_NO_CDROM_SUPPORT 1
+ #else
+ #  error Unknown platform.
+ #endif
+diff -Naur physfs-3.0.2/src/physfs_platform_switch.c physfs-3.0.2-patched/src/physfs_platform_switch.c
+--- physfs-3.0.2/src/physfs_platform_switch.c	1970-01-01 01:00:00.000000000 +0100
++++ physfs-3.0.2-patched/src/physfs_platform_switch.c	2020-05-21 15:29:53.178459807 +0100
+@@ -0,0 +1,106 @@
++/*
++ * Nintendo Switch support routines for PhysicsFS.
++ *
++ * Please see the file LICENSE.txt in the source's root directory.
++ *
++ */
++
++
++#define __PHYSICSFS_INTERNAL__
++#include "physfs_platforms.h"
++
++#ifdef PHYSFS_PLATFORM_SWITCH
++
++#include <switch.h>
++
++// for getcwd
++#include <unistd.h>
++// for PATH_MAX
++#include <limits.h>
++
++#include "physfs_internal.h"
++
++int __PHYSFS_platformInit(void)
++{
++	return 1;  /* always succeed. */
++} /* __PHYSFS_platformInit */
++
++
++void __PHYSFS_platformDeinit(void)
++{
++	/* no-op */
++} /* __PHYSFS_platformDeinit */
++
++
++void __PHYSFS_platformDetectAvailableCDs(PHYSFS_StringCallback cb, void *data)
++{
++	/* no-op */
++}
++
++char *__PHYSFS_switchCalcUserDir(void)
++{
++	/* Use the jail directory (hopefully) found before. */
++	return __PHYSFS_strdup(PHYSFS_getBaseDir());
++}
++
++char *__PHYSFS_platformCalcBaseDir(const char *argv0)
++{
++	char *retval = NULL;
++	/* As there is no system-specific directory, directly inspect argv0. */
++	if (argv0 == NULL)
++	{
++		/* User did not provide a path, just use the current working directory.
++		 *  As physfs should be initialized soon after application start, this
++		 *  should give us a useable directory.
++		 */
++		char fullpath[PATH_MAX];
++		if (getcwd(fullpath, sizeof(fullpath)) != NULL)
++		{
++			const size_t cwdlen = strlen(fullpath);
++			/* getcwd does not provide a trailing slash, add it. */
++			retval = (char*) allocator.Malloc(cwdlen + 2);
++			BAIL_IF(!retval, PHYSFS_ERR_OUT_OF_MEMORY, NULL);
++			strncpy(retval, fullpath, cwdlen);
++			retval[cwdlen] = '/';
++			retval[cwdlen + 1] = '\0';
++		}
++	}
++	else
++	{
++		/* nx-hbmenu should give us the full path of the application, this may
++		 *  reside in a subfolder. Higher level code will strip away the name
++		 *  and extension.
++		 */
++		return NULL;
++	}
++
++	if (!retval)
++		/* Last resort: use `/switch` directory. */
++		retval = __PHYSFS_strdup("sdmc:/switch/");
```

Comment:
Change this to just "/switch/" - avoiding explicit device names will allow code to run from different devices in the future. Is the trailing slash needed?

Reply:
Yes, trailing slash is added everywhere else as well.
> Platform layer is required to append a dirsep